src/main/java/core/basesyntax/interfaces/FruitOperationImpl.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.interfaces;
+
+import core.basesyntax.FruitRecord;
+import java.util.Map;
+
+public class FruitOperationImpl implements FruitOperation {
+
+    @Override
+    public void process(FruitRecord record, Map<String, Integer> stock) {
+
+        stock.putIfAbsent(record.getFruit(), 0);
+        int current = stock.get(record.getFruit());
+
+        switch (record.getOperation()) {
+            case "b":
+                stock.put(record.getFruit(), record.getQuantity());
+                break;
+            case "s":
+            case "r":
+                stock.put(record.getFruit(), current + record.getQuantity());
+                break;
+            case "p":
+                stock.put(record.getFruit(), current - record.getQuantity());
+                break;
+            default:
+                System.out.println("There is a problem with data!");
```

Default case prints a message to stdout. This is not acceptable: on invalid operation codes you must throw a RuntimeException with details (operation code, record content). The checklist notes to rethrow informative RuntimeException on invalid data and to not ignore exceptions or use prints for error handling. Replace System.out.println with throwing RuntimeException. "Don't ignore exceptions in try-catch blocks ... rethrow a RuntimeException with an **informative** message and exception object."